File: risk-lib/src/test/java/net/mjduffin/risk/lib/entities/DiceManagerTest.java
package net.mjduffin.risk.lib.entities;

import net.mjduffin.risk.lib.usecase.GameplayException;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.mockito.Mockito;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.mockito.ArgumentMatchers.any;
import static org.mockito.Mockito.when;

class DiceManagerTest {
    DiceManager diceManager;

    @BeforeEach
    public void setup() {


    }

    @Test
    public void diceTest() throws GameplayException {
        DieThrow dt = Mockito.mock(DieThrow.class);
        when(dt.getDieValue()).thenReturn(1);
        DiceManager diceManager = new DiceManager(dt);

        DiceManager.Result result = diceManager.engage(3, 2);
        assertEquals(1, result.attackers);
        assertEquals(2, result.defenders);
    }

    @Test
    public void diceAttackerWins() throws GameplayException {
        DieThrow dt = Mockito.mock(DieThrow.class);
        when(dt.getDieValue()).thenReturn(6).thenReturn(6).thenReturn(6).thenReturn(4).thenReturn(5);
        DiceManager diceManager = new DiceManager(dt);

        DiceManager.Result result = diceManager.engage(3, 2);
        assertEquals(3, result.attackers);
        assertEquals(0, result.defenders);
    }

}

File: risk-lib/src/test/java/net/mjduffin/risk/lib/entities/DiceManagerTest.kt
package net.mjduffin.risk.lib.entities

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import org.mockito.Mockito

internal class DiceManagerTest {

    @Test
    fun diceTest() {
        val dt = Mockito.mock(DieThrow::class.java)
        Mockito.`when`(dt.dieValue).thenReturn(1)
        val diceManager = DiceManager(dt)
        val result = diceManager.engage(3, 2)
        Assertions.assertEquals(1, result.attackers)
        Assertions.assertEquals(2, result.defenders)
    }

    @Test
    fun diceAttackerWins() {
        val dt = Mockito.mock(DieThrow::class.java)
        Mockito.`when`(dt.dieValue).thenReturn(6).thenReturn(6).thenReturn(6).thenReturn(4).thenReturn(5)
        val diceManager = DiceManager(dt)
        val result = diceManager.engage(3, 2)
        Assertions.assertEquals(3, result.attackers)
        Assertions.assertEquals(0, result.defenders)
    }
}